pkg/agent/flowexporter/connections/conntrack_connections.go

```diff
@@ -116,7 +116,10 @@ func (cs *ConntrackConnectionStore) Poll() ([]int, error) {
 		return nil
 	}
 
-	if err := cs.ForAllConnectionsDo(deleteIfStaleOrResetConn); err != nil {
+	// Hold the lock until verify whether the connection exist in conntrack table.
+	cs.AcquireConnStoreLock()
```

Comment:
I think it could be an issue that we hold the lock for `cs.connDumper.DumpFlows(zone)`. This is a very expensive operation and we tried to avoid doing that in the past IIRC. Would it be possible to call it first, and then acquire the lock?

Reply:
thanks for the great suggestion. I've updated accordingly.